egs/tedlium/s5/local/nnet3/run_tdnn_discriminative.sh

```diff
@@ -0,0 +1,211 @@
+#%WER 13.5 | 507 17792 | 89.1 8.5 2.4 2.6 13.5 86.6 | -0.195 | exp/nnet3/tdnn_sp_smbr/decode_devepoch2.adj/score_11_1.0/ctm.filt.filt.sys
+#%WER 12.6 | 507 17792 | 89.9 7.4 2.6 2.5 12.6 85.2 | -0.241 | exp/nnet3/tdnn_sp_smbr/decode_devepoch2.adj_rescore/score_14_0.5/ctm.filt.filt.sys
+#%WER 13.7 | 507 17792 | 88.8 8.3 2.8 2.6 13.7 87.4 | -0.118 | exp/nnet3/tdnn_sp_smbr/decode_devepoch3.adj/score_13_1.0/ctm.filt.filt.sys
+#%WER 12.6 | 507 17792 | 89.8 7.5 2.7 2.4 12.6 85.0 | -0.307 | exp/nnet3/tdnn_sp_smbr/decode_devepoch3.adj_rescore/score_12_1.0/ctm.filt.filt.sys
+#%WER 13.7 | 507 17792 | 88.8 8.3 2.9 2.5 13.7 87.2 | -0.182 | exp/nnet3/tdnn_sp_smbr/decode_devepoch4.adj/score_14_1.0/ctm.filt.filt.sys
+#%WER 12.8 | 507 17792 | 90.0 7.7 2.3 2.8 12.8 86.0 | -0.300 | exp/nnet3/tdnn_sp_smbr/decode_devepoch4.adj_rescore/score_13_0.5/ctm.filt.filt.sys
+#%WER 12.7 | 1155 27512 | 89.3 8.2 2.5 2.0 12.7 81.0 | -0.151 | exp/nnet3/tdnn_sp_smbr/decode_test/score_12_1.0/ctm.filt.filt.sys
+#%WER 12.5 | 1155 27512 | 89.4 8.1 2.5 1.9 12.5 80.3 | -0.119 | exp/nnet3/tdnn_sp_smbr/decode_testepoch1.adj/score_12_1.0/ctm.filt.filt.sys
+#%WER 11.4 | 1155 27512 | 90.6 7.0 2.3 2.0 11.4 78.9 | -0.184 | exp/nnet3/tdnn_sp_smbr/decode_testepoch1.adj_rescore/score_12_0.5/ctm.filt.filt.sys
+#%WER 12.5 | 1155 27512 | 89.4 8.0 2.6 1.9 12.5 80.5 | -0.120 | exp/nnet3/tdnn_sp_smbr/decode_testepoch2.adj/score_13_1.0/ctm.filt.filt.sys
+#%WER 11.3 | 1155 27512 | 90.4 7.0 2.6 1.7 11.3 79.2 | -0.224 | exp/nnet3/tdnn_sp_smbr/decode_testepoch2.adj_rescore/score_13_1.0/ctm.filt.filt.sys
+#%WER 12.7 | 1155 27512 | 89.3 8.2 2.5 2.0 12.7 81.0 | -0.151 | exp/nnet3/tdnn_sp_smbr/decode_testepoch3.adj/score_12_1.0/ctm.filt.filt.sys
+#%WER 11.5 | 1155 27512 | 90.3 7.1 2.6 1.8 11.5 79.5 | -0.246 | exp/nnet3/tdnn_sp_smbr/decode_testepoch3.adj_rescore/score_13_1.0/ctm.filt.filt.sys
+#%WER 12.8 | 1155 27512 | 89.2 8.2 2.6 2.0 12.8 81.7 | -0.181 | exp/nnet3/tdnn_sp_smbr/decode_testepoch4.adj/score_13_1.0/ctm.filt.filt.sys
+#%WER 11.5 | 1155 27512 | 90.4 7.1 2.5 2.0 11.5 80.6 | -0.248 | exp/nnet3/tdnn_sp_smbr/decode_testepoch4.adj_rescore/score_16_0.5/ctm.filt.filt.sys
+
+
+# Results:
+
+. cmd.sh
+
+
+stage=1
+train_stage=-10 # can be used to start training in the middle.
+get_egs_stage=-10
+use_gpu=true  # for training
+cleanup=false  # run with --cleanup true --stage 6 to clean up (remove large things like denlats,
+               # alignments and degs).
+
+. cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+srcdir=exp/nnet3/tdnn_sp
+train_data_dir=data/train_sp_hires
+online_ivector_dir=exp/nnet3/ivectors_train_sp
+degs_dir=                     # If provided, will skip the degs directory creation
+lats_dir=                     # If provided, will skip denlats creation
+
+## Objective options
+criterion=smbr
+one_silence_class=true
+
+dir=${srcdir}_${criterion}
+
+## Egs options
```

Comment:
I'm actually OK with these variables as long as there are just a few of
them and they are carefully chosen, e.g. they point out a couple of the
more commonly-tuned options.    I this case I think it's OK.

On Sat, Apr 2, 2016 at 5:33 PM, Vijayaditya Peddinti <
notifications@github.com> wrote:

> In egs/tedlium/s5/local/nnet3/run_tdnn_discriminative.sh
> <https://github.com/kaldi-asr/kaldi/pull/644#discussion_r58300230>:
>
> > +. ./path.sh
> > +. ./utils/parse_options.sh
> > +
> > +srcdir=exp/nnet3/tdnn_sp
> > +train_data_dir=data/train_sp_hires
> > +online_ivector_dir=exp/nnet3/ivectors_train_sp
> > +degs_dir=                     # If provided, will skip the degs directory creation
> > +lats_dir=                     # If provided, will skip denlats creation
> > +
> > +## Objective options
> > +criterion=smbr
> > +one_silence_class=true
> > +
> > +dir=${srcdir}_${criterion}
> > +
> > +## Egs options
>
> remove any variable which is only used once.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/644/files/82caa06f43641485acb81b204e02eea21e5bbbeb#r58300230>
>